src/libDirectoryService/DirectoryService.h

```diff
@@ -94,6 +94,11 @@ class DirectoryService : public Executable, public Broadcastable
     std::unordered_map<uint64_t, std::vector<std::vector<unsigned char>>>
         m_MBSubmissionBuffer;
 
+    std::mutex m_mutexFinalBlockConsensusBuffer;
+    std::unordered_map<uint64_t,
```

Comment:
Actually the first key for epoch num is not essential, can be removed

Reply:
@kaikawaliu I will remove it.